src/app/component-library/components/checkbox/checkbox.component.ts

```diff
@@ -20,7 +20,16 @@ export class CheckboxComponent {
   toggleCheck() {
     if (!this.disabled) {
       this.isChecked = !this.isChecked;
-      this.isCheckedChange.emit(this.value);
+      this.isCheckedChange.emit(this.isChecked);
+    }
+  }
+
+  handleKeyDown(event: KeyboardEvent) {
```

Comment:
Suggestion: While this does make keyboard functionality testable in automated tests, I do want to mention that if we use a [native checkbox element](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/checkbox_role#best_practices) that is simply restyled (or hidden with CSS), we do not have to write the checkbox logic ourselves and we don't have to write tests.